src/TwigComponent/src/DependencyInjection/Compiler/TwigComponentPass.php

```diff
@@ -27,23 +28,31 @@ public function process(ContainerBuilder $container): void
     {
         $serviceIdMap = [];
 
-        foreach (array_keys($container->findTaggedServiceIds('twig.component')) as $serviceId) {
-            $definition = $container->getDefinition($serviceId);
+        foreach ($container->getDefinitions() as $id => $definition) {
```

Comment:
I think you already know, but just in case: if you use `registerAttributeForAutoconfiguration`, you can keep this implementation as it was (with `$container->findTaggedServiceIds('twig.component')`)

Reply:
BTW an easy way to implement this, is the following in `TwigComponentExtension.php`:

```php
// Unfortunately only works for PHP8 attributes, not for doctrine annotations
$container->registerAttributeForAutoconfiguration(TwigComponent::class, static function (ChildDefinition $definition, Component $attribute): void {
    $definition->addTag('twig.component', get_object_vars($attribute));
});
```

You would need to make the `TwigComponent` properties public I think and you could also remove the getters that way.

It's done the same way in `FrameworkExtension.php` (symfony/framework-bundle):

```php
$container->registerAttributeForAutoconfiguration(AsEventListener::class, static function (ChildDefinition $definition, AsEventListener $attribute): void {
    $definition->addTag('kernel.event_listener', get_object_vars($attribute));
});
```

Reply:
Thanks! I opted to set the tag parameters manually. `name` is reserved for the tag's name in XML.